src/bblocks_data_importers/protocols.py

```diff
@@ -16,6 +16,6 @@ class DataImporter(Protocol):
     - get_data: get the data from the source.
     """
 
-    def get_data(self) -> pd.DataFrame:
+    def get_data(self, *args, **kwargs) -> pd.DataFrame:
```

Comment:
This caused IDE warnings `Type of 'get_data' is incompatible with 'DataImporter' `. How should we handle this?

Reply:
I get it in every implementation of the protocol. I'm not sure about what causes it tbh.... Maybe an ABC would stop the warnings?

Reply:
the addition of *args and **kwargs caused it for all the importers, where before I wasn't getting any warning. It resolves if you add *args and **kwargs to all the implementations of `get_data` so that the parameters match

Reply:
so what should we do?

Reply:
Different options. We have so far been inheriting the protocol which is fine because it doesn't provide any provide any concrete functionality. If we keep it as a base class then we need to make the implementation of `get_data` as flexible as the protocol i.e. have *args and **kwargs in the method. Alternatively, we don't need this explicit inheritance, and we can just make sure that all the importers follow the protocol structure. In the spirit of flexibility I'd suggest the second option but we need to be conscious about the functionality when we make new importers so that there is some consistency